pkg/prometheus/server/statefulset_test.go

```diff
@@ -2906,3 +2906,63 @@ func TestPersistentVolumeClaimRetentionPolicy(t *testing.T) {
 		t.Fatalf("expected persistentVolumeClaimDeletePolicy.WhenScaled to be %s but got %s", appsv1.DeletePersistentVolumeClaimRetentionPolicyType, sset.Spec.PersistentVolumeClaimRetentionPolicy.WhenScaled)
 	}
 }
+
+func TestStartupProbeTimeoutSeconds(t *testing.T) {
+	tests := []struct {
+		maximumStartupDurationSeconds   *int32
+		expectedStartupPeriodSeconds    int32
+		expectedStartupFailureThreshold int32
+	}{
+		{
+			maximumStartupDurationSeconds:   nil,
+			expectedStartupPeriodSeconds:    15,
+			expectedStartupFailureThreshold: 60,
+		},
+		{
+			maximumStartupDurationSeconds:   ptr.To(int32(0)),
+			expectedStartupPeriodSeconds:    15,
+			expectedStartupFailureThreshold: 60,
+		},
+		{
+			maximumStartupDurationSeconds:   ptr.To(int32(600)),
+			expectedStartupPeriodSeconds:    10,
+			expectedStartupFailureThreshold: 60,
+		},
+		{
+			maximumStartupDurationSeconds:   ptr.To(int32(900)),
+			expectedStartupPeriodSeconds:    15,
+			expectedStartupFailureThreshold: 60,
+		},
+		{
+			maximumStartupDurationSeconds:   ptr.To(int32(1200)),
+			expectedStartupPeriodSeconds:    20,
+			expectedStartupFailureThreshold: 60,
+		},
+		{
+			maximumStartupDurationSeconds:   ptr.To(int32(129)),
+			expectedStartupPeriodSeconds:    2,
+			expectedStartupFailureThreshold: 64,
```

Comment:
I'd expect expectedStartupPeriodSeconds * expectedStartupFailureThreshold to be > 129.

Reply:
Actually it will not be, because go round down integer divisions
129 / 60 (converting to second) = 2 (instead of 2.15 because of truncation division)

Reply:
yes I understand and I'd prefer if the value was rounded up to the next integer: as a user if I specify "maximumStartupDurationSeconds: 170", I don't expect the probe to fail after only 120s.